### PR TITLE
Fixing merge

### PR DESCRIPTION
**Pull request by Ferotiq**
### Description

Fetching the upstream for the typescript branch

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,11 +11,13 @@
...
-    "build-ts": "npx gulp build-ts",
-    "start:dev-ts": "npm run build-ts && concurrently --kill-others \"npx gulp watch-ts\" \"nodemon --watch ./backend ./backend/server.js\" \"firebase serve --only hosting\"",
-    "start:dev:nodb-ts": "npm run build-ts && concurrently --kill-others \"npx gulp watch-ts\" \"firebase serve --only hosting\"",
-    "start:dev:nofe-ts": "nodemon --watch ./backend ./backend/server.js",
-    "deploy:live-ts": "npm run build-ts && firebase deploy -P live --only hosting"
+    "start:dev-ts": "concurrently --kill-others \"cd ./frontend && npm run start:dev\" \"cd ./backend && npm run start:dev\"",
+    "start:dev:be-ts": "cd backend && npm run start:dev-ts",
+    "start:dev:fe-ts": "cd frontend && npm run start:dev-ts",
+    "deploy:live-ts": "cd frontend && npm run deploy:live-ts",
+    "install:all-ts": "sh ./bin/install.sh",
```
Comment (Bruception) [inline package.json:18]: I think we can remove this script. The install script just installs all the dependencies for the project.

Comment (Ferotiq) [reply package.json]: fixed

**Comment by Bruception** [comment]
lgtm iiw

File: package.json [hunk 2]
```diff
@@ -11,11 +11,12 @@
     "install:all": "sh ./bin/install.sh",
     "lint": "./node_modules/.bin/eslint './backend/**/*.js' './frontend/src/js/**/*.js'",
     "pretty": "prettier --check './backend/**/*.js' './frontend/src/**/*.{js,scss}' './frontend/static/**/*.{json,html}'",
-    "build-ts": "npx gulp build-ts",
-    "start:dev-ts": "npm run build-ts && concurrently --kill-others \"npx gulp watch-ts\" \"nodemon --watch ./backend ./backend/server.js\" \"firebase serve --only hosting\"",
-    "start:dev:nodb-ts": "npm run build-ts && concurrently --kill-others \"npx gulp watch-ts\" \"firebase serve --only hosting\"",
-    "start:dev:nofe-ts": "nodemon --watch ./backend ./backend/server.js",
-    "deploy:live-ts": "npm run build-ts && firebase deploy -P live --only hosting"
+    "start:dev-ts": "concurrently --kill-others \"cd ./frontend && npm run start:dev\" \"cd ./backend && npm run start:dev\"",
```
Comment (Bruception) [inline package.json:14]: This script needs to be: `"concurrently --kill-others \"cd ./frontend && npm run start:dev-ts\" \"cd ./backend && npm run start:dev-ts\""`

File: package.json [hunk 3]
```diff
@@ -11,11 +11,12 @@
...
+    "start:dev:be-ts": "cd backend && npm run start:dev-ts",
+    "start:dev:fe-ts": "cd frontend && npm run start:dev-ts",
+    "deploy:live-ts": "cd frontend && npm run deploy:live-ts",
+    "lint-ts": "./node_modules/.bin/eslint './backend/**/*.ts' './frontend/src/ts/**/*.ts'",
+    "pretty-ts": "prettier --check './backend/**/*.ts' './frontend/src/**/*.{ts,scss}' './frontend/static/**/*.{json,html}'"
```
Comment (Bruception) [inline package.json:19]: We can remove this command and just update the pretty command to be:
`"prettier --check './backend/**/*.{js,ts}' './frontend/src/**/*.{js,ts,scss}' './frontend/static/**/*.{json,html}'"`

File: package.json [hunk 4]
```diff
@@ -11,16 +11,21 @@
...
-    "build-ts": "npx gulp build-ts",
-    "start:dev-ts": "npm run build-ts && concurrently --kill-others \"npx gulp watch-ts\" \"nodemon --watch ./backend ./backend/server.js\" \"firebase serve --only hosting\"",
-    "start:dev:nodb-ts": "npm run build-ts && concurrently --kill-others \"npx gulp watch-ts\" \"firebase serve --only hosting\"",
-    "start:dev:nofe-ts": "nodemon --watch ./backend ./backend/server.js",
-    "deploy:live-ts": "npm run build-ts && firebase deploy -P live --only hosting"
+    "start:dev-ts": "concurrently --kill-others \"cd ./frontend && npm run start:dev-ts\" \"cd ./backend && npm run start:dev-ts\"",
+    "start:dev:be-ts": "cd backend && npm run start:dev-ts",
+    "start:dev:fe-ts": "cd frontend && npm run start:dev-ts",
+    "deploy:live-ts": "cd frontend && npm run deploy:live-ts",
+    "lint-ts": "./node_modules/.bin/eslint './backend/**/*.ts' './frontend/src/ts/**/*.ts'",
```
Comment (Bruception) [inline package.json:18]: We can also remove this command.
Update lint to: `"./node_modules/.bin/eslint './backend/**/*.{js,ts}' './frontend/src/**/*.{js,ts}'"`

File: package.json (hunk 3 above)
Comment (Ferotiq) [reply package.json]: fixed

File: package.json (hunk 2 above)
Comment (Ferotiq) [reply package.json]: fixed

File: package.json (hunk 4 above)
Comment (Ferotiq) [reply package.json]: fixed

File: package.json [hunk 5]
```diff
@@ -10,17 +10,21 @@
...
+    "start:dev:be-ts": "cd backend && npm run start:dev-ts",
+    "start:dev:fe-ts": "cd frontend && npm run start:dev-ts",
+    "deploy:live-ts": "cd frontend && npm run deploy:live-ts",
+    "lint-ts": "./node_modules/.bin/eslint './backend/**/*.ts' './frontend/src/ts/**/*.ts'"
   },
   "engines": {
     "npm": "8.1.2"
   },
   "devDependencies": {
+    "@types/chart.js": "^2.9.35",
```
Comment (Bruception) [inline package.json:23]: Let's move these dependencies to the frontend package file.

File: package.json [hunk 6]
```diff
@@ -10,17 +10,21 @@
...
+    "@types/chartjs-plugin-annotation": "^0.5.2",
```
Comment (Bruception) [inline package.json:24]: Same as above

File: package.json [hunk 7]
```diff
@@ -31,6 +34,7 @@
     "eslint-plugin-promise": "5.1.0",
     "eslint-plugin-require-path-exists": "1.1.9",
     "husky": "4.3.0",
+    "ms": "^2.1.3",
```
Comment (Bruception) [inline package.json:37]: This too.

File: package.json (hunk 5 above)
Comment (Ferotiq) [reply package.json:23]: It won't work without it in the parent directory.

**Comment by Bruception** [comment]
RE-LGTM IIW

**Review by Miodec — APPROVED** [review]

**Pull request merged**